docs/transforms.md

```diff
@@ -0,0 +1,114 @@
+## Transforms
+
+Transforms are operations applied on single records (i.e. multi-message transforms are not yet supported). 
+
+They are optional parts of a pipeline and, if present, they come in the form of processors (more information about 
+processor can be found in the [Architecture documentation](/docs/architecture.md)). As processors, transforms can be 
+attached either to a connector or to a pipeline. In other words, transforms can transform:
+1. Records coming out of a source connector (if the transform is attached to a source connector)
+2. Records coming into a destination connector (if the transform is attached to a destination connector)
+3. All records flowing through a pipeline (if the transform is attached to a pipeline)
+
+We'll go a little deeper into the case of a pipeline transform. At a high-level, source connectors, pipeline transforms
+and destination connector are laid out as in this image:
+
+![Pipeline](data/pipeline.svg)
+
+The main point here is that all pipeline transforms are "grouped together". Records from all sources will be sent to the
+transform in processor 1, then processor 2 etc. Pipeline transforms are chained in the order in which they were created.
+
+## Adding and configuring a transform
+
+Since a transform is a type of processor, all transforms are created through the `processors` endpoint. 
+Here's an example:
+
+```json lines
+POST /v1/processors
+{
+    // name of the transform
+    "name": "extractfieldpayload",
+    // type of processor: 1 - transform, 2 - filter 
+    "type": 1,
+    "parent": 
+    {
+        // type of parent: 1 - connector, 2 - pipeline
+        "type": 1,
+        // parent ID (connector ID in this case)
+        "id": "aed07589-44d8-4c68-968c-1f6c5197f13b" 
+    },
+    "config":
+    {
+        "settings":
+        {
+            // configuration map for this processor
+            "field": "name" 
+        }
+    }
+}
+```
+The request to create a processor is described in [api.proto](/proto/api/v1/api.proto).
+
+## Supported transforms
+
+Conduit provides a number of built-in transforms, such as filtering fields, replacing them, posting payloads to HTTP endpoints etc.
+Conduit also provides the ability to write custom transforms in JavaScript.
+
+### Built-in transforms
+
+An up-to-date list of all built-in transforms and detailed descriptions can be found [here](https://pkg.go.dev/github.com/conduitio/conduit/pkg/processor/transform/txfbuiltin).
+An example is available in [extract-field-transform.sh](/examples/transforms/extract-field-transform.sh). The script will
+set up a pipeline with the built-in extract-field transform.
+
+### JavaScript transforms
+
+JavaScript transforms make it possible to write custom transforms. The API name for JavaScript transforms (used in the
+request to create a transform) is `js`. There's only one configuration parameters, `script`, which is the script itself.
```

Comment:
I think this would be:

```suggestion
request to create a transform) is `js`. There's only one configuration parameter, `script`, which is the script itself.
```

Reply:
Good one!